packages/graphback-codegen-client/src/plugin/templates/fragmentFields.ts

```diff
@@ -0,0 +1,54 @@
+import { getNamedType, GraphQLObjectType, isCompositeType, isWrappingType } from 'graphql';
+
+/**
+ * For given type - it returns list of the type fields that 
+ * can be used for building GraphQL Fragment
+ * 
+ * 
+ * @param t type that is going to be used 
+ * @param level nested level (supports only 0, 1 nested level)
+ */
+export function buildReturnFields(t: GraphQLObjectType, level?: number) {
+    const fieldsMap = t.getFields();
+    if (level > 1) {
+      throw new Error("Function supports only 1 nested level")
+    }
+  
+    // tslint:disable-next-line: typedef
+    return Object.keys(fieldsMap).reduce((data, key) => {
+      const field = fieldsMap[key];
+      if (isWrappingType(field.type)) {
+        const modelType = getNamedType(field.type);
+        if (isCompositeType(modelType)) {
+          if (level !== 0) {
+            data.push({
+              [field.name]: buildReturnFields(modelType as GraphQLObjectType, level - 1)
+            });
+          }
+
+          // Nested relation that should not be included
+          return data;
+        }
+      }
+      data.push(field.name);
+  
+      return data;
+    }, []);
+  }
+  
+  export const printReturnFields = (resultArray: any[], shift: string = '') => {
+    let resultString = '';
+  
+    for (const element of resultArray) {
+      if (element instanceof Object) {
+        const key = Object.keys(element)[0];
+        resultString += `   ${key} {
+${printReturnFields(element[key], '   ')}
+   }`
+      } else {
+        resultString += `${shift}   ${element}\n`;
```

Comment:
Causes a blank newline at end of fragment. We should leave this though and have a common formatter than can be used by plugins.

Reply:
What do you mean by common formatter. What fix I can apply here?

Reply:
Going to fix it outside \n stays here :D